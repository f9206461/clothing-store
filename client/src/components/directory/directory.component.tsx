import React, { Fragment } from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import { selectDirectorySections } from "../../redux/directory/directory.selector";

import { Title, DirectoryMenuCont } from "./directory.styles";
import MenuItem from '../menu-item/menu-item.component'

export const Directory = ({ sections }: { sections: Section[] }) => {
  return (
    <Fragment>
      <Title>Welcome to CRWN Clothing!</Title>
      <DirectoryMenuCont>
          {
              sections.map(({id, ...otherSectionProps}) => {
                  return <MenuItem key={id} {...otherSectionProps}/>
              })
          }
      </DirectoryMenuCont>
    </Fragment>
  )
}

const mapStateToProps = createStructuredSelector({
  sections: selectDirectorySections
})

export default connect(mapStateToProps)(Directory);