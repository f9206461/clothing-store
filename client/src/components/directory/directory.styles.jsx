import styled from "styled-components";

export const DirectoryMenuCont = styled.div`
    width: 100%;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
`;

export const Title = styled.h2`
    font-size: 38px;
    margin: 0 auto 30px;
    margin-bottom: 60px;

    @media screen and (max-width: 800px) {
        font-size: 30px;
        margin-bottom: 30px;
    }
`;