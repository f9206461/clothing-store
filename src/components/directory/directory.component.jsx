import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import { selectDirectorySections } from "../../redux/directory/directory.selector";

import { DirectoryMenuCont } from "./directory.styles";
import MenuItem from '../menu-item/menu-item.component'

const Directory = ( { sections } ) => {
  return (
    <DirectoryMenuCont>
        {
            sections.map(({id, ...otherSectionProps}) => {
                return <MenuItem key={id} {...otherSectionProps}/>
            })
        }
    </DirectoryMenuCont>
  )
}

const mapStateToProps = createStructuredSelector({
  sections: selectDirectorySections
})

export default connect(mapStateToProps)(Directory);