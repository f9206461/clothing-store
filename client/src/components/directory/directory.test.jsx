import React from "react";
import { shallow } from "enzyme";

// Import the Directory component without CONNECT (not default)
import { Directory } from './directory.component';

/*
What to test:
1. Render Directory component correctly.
2. Check if the rendered MenuItem match the number of section in props.
*/

describe('Directory Component', () => {
    let wrapper;

    beforeEach(() => {
        const mockProps = {
            sections: [
                { id: 1, title: 'hats' },
                { id: 2, title: 'jackets' }
            ]
        }
        wrapper = shallow(<Directory {...mockProps} />);
    });

    it('should render Directory component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the correct number of MenuItem component', () => {
        expect(wrapper.find('withRouter(MenuItem)').length).toEqual(2);
    });
})