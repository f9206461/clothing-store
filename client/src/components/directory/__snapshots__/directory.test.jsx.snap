// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory Component should render Directory component 1`] = `
<Fragment>
  <Title>
    Welcome to CRWN Clothing!
  </Title>
  <DirectoryMenuCont>
    <withRouter(MenuItem)
      key="1"
      title="hats"
    />
    <withRouter(MenuItem)
      key="2"
      title="jackets"
    />
  </DirectoryMenuCont>
</Fragment>
`;
